components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManagerTest.java

```diff
@@ -157,6 +175,667 @@ public void testRetrieveUserRecoveryInformation() throws Exception {
+            });
+        } else if ("getUserListByClaims".equals(methodName)) {
+            assertThrows(IdentityRecoveryException.class, () -> {
+                userAccountRecoveryManager.getUserListByClaims(userClaims, MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            });
+        }
+    }
+
+    @DataProvider(name = "getUserByClaimsExceptions")
+    public Object[][] exceptionScenarios() {
+
+        return new Object[][] {
+                { "getUsernameByClaims" },
+                { "getUserListByClaims" }
+        };
+    }
+
+    /**
+     * Tests that an IdentityRecoveryServerException is thrown during GetClaimListOfUser
+     *
+     * @throws Exception if there is an issue with mocking or method invocation.
+     */
+    @Test (expectedExceptions = IdentityRecoveryServerException.class)
+    public void testGetClaimListOfUserException() throws Exception {
+
+        mockUserstoreManager();
+        mockBuildUser();
+        mockedUtils.when(() -> Utils.isNotificationsInternallyManaged(anyString(), any()))
+                .thenReturn(true);
+        when(identityRecoveryServiceDataHolder.getMultiAttributeLoginService())
+                .thenReturn(multiAttributeLoginService);
+        when(multiAttributeLoginService.isEnabled(anyString())).thenReturn(false);
+        when(abstractUserStoreManager.getUserListWithID(any(Condition.class), anyString(), anyString(),
+                anyInt(), anyInt(), isNull(), isNull())).thenReturn(getOneFilteredUser());
+        when(claimManager.getAttributeName(anyString(), anyString()))
+                .thenReturn("http://wso2.org/claims/mockedClaim");
+        mockedIdentityUtil.when(IdentityUtil::getPrimaryDomainName).thenReturn("PRIMARY");
+        when(abstractUserStoreManager.getUserClaimValues(anyString(), any(String[].class), isNull()))
+                .thenThrow(new UserStoreException("Simulated exception"));
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(IdentityEventConstants.EventProperty.USER, "user");
+        when(IdentityRecoveryServiceDataHolder.getInstance().getIdentityEventService())
+                .thenReturn(identityEventService);
+        doThrow(new IdentityEventException("error"))
+                .when(identityEventService).handleEvent(any(Event.class));
+        when(Utils.handleServerException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_ERROR_LOADING_USER_CLAIMS, null))
+                .thenReturn(new IdentityRecoveryServerException(null, null, null));
+        userAccountRecoveryManager.retrieveUserRecoveryInformation(userClaims, StringUtils.EMPTY,
+                RecoveryScenarios.USERNAME_RECOVERY, properties);
+    }
+
+    /**
+     * Tests testGetDomainNames through retrieveUserRecoveryInformation
+     *
+     * @throws Exception if there is an issue with mocking or method invocation.
+     */
+    @Test(expectedExceptions = NullPointerException.class)
```

Comment:
Is NPE intentional?

Reply:
https://github.com/wso2-extensions/identity-governance/pull/859/files#r2041072832